crypto/keyring/info.go

```diff
@@ -191,21 +195,14 @@ type multiInfo struct {
 }
 
 // NewMultiInfo creates a new multiInfo instance
-func NewMultiInfo(name string, pub cryptotypes.PubKey) Info {
-	multiPK := pub.(*multisig.LegacyAminoPubKey)
-
-	pubKeys := make([]multisigPubKeyInfo, len(multiPK.PubKeys))
-	for i, pk := range multiPK.GetPubKeys() {
-		// TODO: Recursively check pk for total weight?
-		pubKeys[i] = multisigPubKeyInfo{pk, 1}
+func NewMultiInfo(name string, pub cryptotypes.PubKey) (Info, error) {
+	if _, ok := pub.(*multisig.LegacyAminoPubKey); !ok {
+		return nil, fmt.Errorf("MultiInfo supports only multisig.LegacyAminoPubKey, got  %T", pub)
 	}
-
 	return &multiInfo{
-		Name:      name,
-		PubKey:    pub,
-		Threshold: uint(multiPK.Threshold),
-		PubKeys:   pubKeys,
-	}
+		Name:   name,
+		PubKey: pub,
+	}, nil
```

Comment:
shouldn't we keep the 2 deleted fields, following https://github.com/cosmos/cosmos-sdk/pull/7477#discussion_r556534909?